ee/clickhouse/client.py

```diff
@@ -133,26 +134,36 @@ def sync_execute(query, args=None, settings=None, with_column_types=False):
         with ch_pool.get_client() as client:
             start_time = perf_counter()
             tags = {}
-            if app_settings.SHELL_PLUS_PRINT_SQL:
-                print()
-                print(format_sql(query, args))
 
-            parameterized_sql, tags = _annotate_tagged_query(query, args)
+            # NOTE: `client.execute` would normally handle substitution, but
+            # because we want to strip the comments to make it easier to copy
+            # and past queries from the `system.query_log` easily with metabase
+            # (metabase doesn't show new lines, so with comments, you can't get
+            # a working query without exporting to csv or similar), we need to
+            # do it manually.
+            #
+            # We only want to try to substitue for SELECT queries, which
+            # clickhouse_driver at this moment in time decides based on the
+            # below predicate.
+            if isinstance(args, (list, tuple, types.GeneratorType)):
```

Comment:
? When is this the case? This doesn't seem sane.

Reply:
Specifically this relates to https://github.com/mymarilyn/clickhouse-driver/blob/master/clickhouse_driver/client.py#L280

I can add a reference to this.

Reply:
It was quite confusing actually. The tests started to fail without this. There are some cases where we're catching all Exceptions like https://github.com/PostHog/posthog/blob/chore/remove-clickhouse-sql-comments/posthog/models/cohort.py#L192:L192

It took me some time to figure out what was going on. I could put in a separate PR to catch more specific errors, or maybe just change to reraise on `settings.TEST == True` as well?